File: src/main/java/com/my/admin/app/commerce/model/document/Product.java
package com.my.admin.app.commerce.model.document;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.io.Serializable;
@Document(collection = "products")
@Data
public class Product implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    private Long id;

    private String name;

    private Double price;

    private String code;

    private ProductType productType;
}

File: src/main/kotlin/com/bk/reactive/app/commerce/admin/my/model/document/Product.kt
package com.bk.reactive.app.commerce.admin.my.model.document

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document(collection = "products")
data class Product(
    @Id
    val id: String,
    val name: String,
    val price: Double,
    val code: String,
    val productType: ProductType,
    val toppings: List<String> = arrayListOf()
)